sys/vm/vm_map.c

```diff
@@ -5140,7 +5178,8 @@ vm_map_lookup_locked(vm_map_t *var_map,		/* IN/OUT */
 	 * Check whether this task is allowed to have this page.
 	 */
 	prot = entry->protection;
-	fault_type &= VM_PROT_READ | VM_PROT_WRITE | VM_PROT_EXECUTE;
+	fault_type &= VM_PROT_READ | VM_PROT_WRITE | VM_PROT_EXECUTE |
+	    VM_PROT_READ_CAP | VM_PROT_WRITE_CAP;
```

Comment:
```suggestion
	fault_type &= VM_PROT_ALL_CAP;
```
Although maybe should upstream VM_PROT_ALL.

Reply:
Yes, I think I'd like to do the upstream change first so that the diff is relative to upstream.